File: backend/src/application/tenant_service.py
from src.domain.tenant.tenant import Tenant, TenantCreate, TenantUpdate
from src.domain.tenant.tenant_repository import TenantRepository
from typing import List, Optional

class TenantService:
    def __init__(self, tenant_repository: TenantRepository):
        self.tenant_repository = tenant_repository

    def create_tenant(self, tenant: TenantCreate) -> Tenant:
        return self.tenant_repository.create(tenant)

    def get_all_tenants(self) -> List[Tenant]:
        return self.tenant_repository.get_all()

    def get_tenant_by_id(self, tenant_id: str) -> Optional[Tenant]:
        return self.tenant_repository.get_by_id(tenant_id)

    def update_tenant(self, tenant_id: str, tenant_update: TenantUpdate) -> Optional[Tenant]:
        return self.tenant_repository.update(tenant_id, tenant_update)

    def delete_tenant(self, tenant_id: str) -> bool:
        return self.tenant_repository.delete(tenant_id)

File: golang/devops_console/internal/application/tenant_service.go
package application

import (
    "devops_console/internal/domain/tenant"
)

type TenantService struct {
    tenantRepository tenant.TenantRepository
}

func NewTenantService(tenantRepository tenant.TenantRepository) *TenantService {
    return &TenantService{tenantRepository: tenantRepository}
}

func (s *TenantService) CreateTenant(tenantCreate tenant.TenantCreate) (*tenant.Tenant, error) {
    return s.tenantRepository.Create(tenantCreate)
}

func (s *TenantService) GetAllTenants() ([]*tenant.Tenant, error) {
    return s.tenantRepository.GetAll()
}

func (s *TenantService) GetTenantByID(tenantID string) (*tenant.Tenant, error) {
    return s.tenantRepository.GetByID(tenantID)
}

func (s *TenantService) UpdateTenant(tenantID string, tenantUpdate tenant.TenantUpdate) (*tenant.Tenant, error) {
    return s.tenantRepository.Update(tenantID, tenantUpdate)
}

func (s *TenantService) DeleteTenant(tenantID string) error {
    return s.tenantRepository.Delete(tenantID)
}